### ISSUE TITLE
Project SDK not set on Modules correctly

### ISSUE THREAD

**Issue by benkeil** [KronicDeth/intellij-elixir#2061]
**Describe the bug**
I don't have auto completion for any base module like `IO`, `Map` or `Enum`, which makes it really hard for a beginner to work with IntelliJ.

**Desktop:**
 - OS: macOS
 - Version 11.4

**Erlang:**
- Installer homebrew
- Version 24

**Elixir:**
- Installer homebrew
- Version 1.12.2

**Plugin:**
- Version 11.13.0

**Comment by KronicDeth** [comment]
Which IDE are you using?  Do you have your SDK comfigured?

**Comment by benkeil** [comment]
I use IntelliJ Ultimate and as far as I see, I configured the SDK.

**Comment by benkeil** [comment]
Is this plugin really maintained?

**Comment by KronicDeth** [comment]
Yes, it’s maintained.  I work on it an hour a night and on Fridays.

I’ll get your bug report when I’m finished on the feature work for the next release.  If you want more priority, GitHub sponsors is available.

**Comment by KronicDeth** [comment]
Post screenshots of your Project Structure dialog with the SDK selected and the SDK selection for each Module.

**Comment by benkeil** [comment]
https://postimg.cc/Lq4hJXwZ

**Comment by KronicDeth** [comment]
Click on the Modules entry under Project and double check the Project SDK is associated with all the modules in the project too.

**Comment by benkeil** [comment]
The problem was that I had configured 2 SDKs in IntelliJ, an actual 1.12 and an older (no longer existent) 1.11. After deleting the non existing SDK and deletion of the `.idea` folder and the `*.iml` file and reopen the project, the correct SDK was set to the modules.

**Comment by sh41** [comment]
Thanks for diagnosing this yourself — a module still pointing at an SDK you had deleted, alongside a healthy project SDK, is exactly the failure, and at the time nothing told you it had happened.

Since v23.9.0-pre-4 (current stable v24.0.1) the plugin detects it. When a module's SDK entry names an SDK that is no longer in the SDK table, the status bar shows an error and a notification names the module and where to fix it. **Tools → Reconfigure Elixir Module Setup** then repairs it, pointing every Elixir Mix module back at the project SDK — the same result your `.idea`/`.iml` deletion got, without the re-import.

It is still not automatic: you have to run that action, or set Module SDK to "Project SDK" in Project Structure yourself. Closing on that basis. If a module ends up on a dead SDK with no warning at all, please reopen.

> Closed as part of an AI-assisted review of the backlog. A human checked this one before closing it, but it might still be a mistake. If you think the reason above is wrong, please reopen the issue and tell us why.
